components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp

```diff
@@ -878,5 +879,79 @@ namespace scream {
             });
         }
 
+        void compute_aerocom_cloudtop(
+            int ncol, int nlay, const real2d &tmid, const real2d &pmid,
+            const real2d &p_del, const real2d &z_del, const real2d &qc,
+            const real2d &qi, const real2d &cldfrac_tot, const real2d &nc,
+            real1d &T_mid_at_cldtop, real1d &p_mid_at_cldtop,
+            real1d &cldfrac_ice_at_cldtop, real1d &cldfrac_liq_at_cldtop,
+            real1d &cldfrac_tot_at_cldtop, real1d &nc_at_cldtop) {
+            // initialize the 1D "clear fraction" as 1 (totally clear) + and
+            // helper variable as 0
+            auto aerocom_clr = real1d("aerocom_clr", ncol);
+            auto aerocom_tmp = real1d("aerocom_tmp", ncol);
+            memset(aerocom_clr, 1);
+            memset(aerocom_tmp, 0);
+            // initialize the 2D "cloudy fraction" as 0 (totally clear)
+            auto aerocom_cld = real2d("aerocom_cld", ncol, nlay);
+            memset(aerocom_cld, 0);
+            // get gravity constant from constants
+            using physconst = scream::physics::Constants<Real>;
+            // Loop over all columns in parallel
+            yakl::fortran::parallel_for(
+                SimpleBounds<1>(ncol), YAKL_LAMBDA(int icol) {
+                  // Loops over all layers in serial (due to accumulative
+                  // product) Start at 2 (second highest) layer because the
+                  // highest one has no clouds
+                  for(int ilay = 2; ilay <= nlay; ++ilay) {
+                    // We only do the calculation if certain conditions are met
+                    if(qc(icol, ilay) + qi(icol, ilay) > 0 &&  // BAD_CONSTANT!
```

Comment:
What's the bad constant in the above line? Are you saying `qc + qi > 0` is a heuristic constraint for deciding whether to perform the calculation?

Reply:
You might consider using variables with descriptive names for these thresholds instead of the magic numbers to make this a little more obvious.

Reply:
Side comment: the syntax here is correct, since `+` has precedence over `&&`. Alas, for a second I was wondering "is this doing + or > first? Is this a bug?", and had to google c++ ops precedence rules (I google that page way more than I should). Perhaps we could use some parentheses to help readers avoid confusion.